block/manager_test.go

```diff
@@ -94,3 +98,94 @@ func TestIsDAIncluded(t *testing.T) {
 	m.blockCache.setDAIncluded(hash.String())
 	require.True(m.IsDAIncluded(hash))
 }
+
+func TestSubmitBlocks(t *testing.T) {
+	require := require.New(t)
+	ctx := context.Background()
+	logger := test.NewFileLoggerCustom(t, test.TempLogFileName(t, t.Name()))
+
+	// Create a minimalistic block manager
+	m := &Manager{
+		dalc:       &da.DAClient{DA: goDATest.NewDummyDA(), GasPrice: -1, Logger: logger},
+		blockCache: NewBlockCache(),
+		logger:     logger,
+	}
+
+	maxDABlobSizeLimit, err := m.dalc.DA.MaxBlobSize(ctx)
+	require.NoError(err)
+
+	testCases := []struct {
+		name                string
+		blocks              []*types.Block
+		isErrExpected       bool
+		expectedNumAttempts uint64
+	}{
+		{
+			name:                "happy path, all blocks A, B, C are submitted on first round",
+			blocks:              []*types.Block{types.GetRandomBlock(1, 5), types.GetRandomBlock(2, 5), types.GetRandomBlock(3, 5)},
+			isErrExpected:       false,
+			expectedNumAttempts: uint64(1),
+		},
+		{
+			name: "blocks A and B are submitted first round because including c triggers blob size limit. C is submitted on second round",
```

Comment:
if we have a separate unit test around blocks not being included once the max size is hit, then that allows this test to rely on that assumption, which allows us to not care about the number of attempts.